Library javascript project Pull request

sergiocortessat opened the pull request:
Hello TSE

Thanks for checking on my project. According to the requirements, here I present the page library which in general terms is able to receive new book entries and display them. as well as to delete or mark them as read

NOTE: This project is presented solo as per authorization of student success to continue Javascript alone.

- The project uses localStorage instead of an empty array.
- The user can click on Add Book which will display the form fields to enter
- User can enter book entries with name, description, and number of pages parameter.
- Once the book is added, the new entries are displayed on a card layout one next to the other
- Each book card entry contains the buttons not read and delete
- The not-read button changes dynamically if the user presses it. the context changes to Read

Commit:
Add index and logo

Commit:
Add simple styling

Commit:
Add function addBookToLibrary

Commit:
Add function Book()

Commit:
Add eventlistener delete button

Commit:
Add readbutton function and eventlistener

Commit:
Add styke to buttons

Commit:
Add proper readme

descholar-ceo commented on README.md:
````diff
@@ -57,56 +55,16 @@ Terminal or similar to execute the program.
-3. run the command ```rails db:create``` This will initialize the necessary migration files
-4. run the command ```rails db:migrate``` 
-5. run the command ```rails server```
+- Clone this repository with git clone ```https://github.com/sergiocortessat/JS_Library-/tree/feature``` using your terminal or command line.
````
- [x] [Optional] You said that we need to clone this repo `https://github.com/sergiocortessat/JS_Library-/tree/feature` whereas this is not the real link to your repo, make sure that you remove everything starting from `/tree`

descholar-ceo reviewed (changes requested):
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @sergiocortessat  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your library project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implement all of the requirements so carefully, keep it up, and your project is really almost done.
:+1: I liked that all of the aspects of the requirements you tried enough to make them,
#### :warning: However we need to make few changes to mark this project approved. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
### :warning: Note: Almost all of the requirements are present up on running the app, all you need is to refactor your functions `addBookToLibrary()` , `Book()` :point_left: Make sure that you respect all of the requirements and that are implemented in a way they are described
- [x] It has an array to store the instances of Book ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/book_array_requirement.png) from Odin)
- [x] It has a constructor ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/book_array_requirement.png) from Odin). An example of constructor that you are require to implement here is like the following:
```javascript
function Book(name, author, pages, read) {
    this.name = name;
    this.author = author;
    this.pages = pages;
    this.read = read;
}

```

- [x] The application has a addBookToLibrary function
- [x] It can take user's input
- [x] It stores the new Book objects into an array
- [x] It has a function that loops through the array and displays each book ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/books_render_loop.png) from Odin)
- [x] It displays on the page every book contained in the array ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/render_requirement.png) from Odin)
- [x] Each Book displayed has a button to remove the Book from the Library array ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/remove_requirement.png) from Odin)
- [x] Each Book displayed has a button to toggle the Book read status ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/library/toggle_requirement.png) from Odin)
- [x] Keep the application logic separated from DOM manipulation tasks ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) from Pathwright)
- [x] Keep all the styling logic in CSS files ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/styling_requirement.png)  from Pathwright)

# [Optional changes] :yellow_circle: 
- [x] Inorder to make your repository professional, you need to make sure that all of the information presented on readme file are accurate, so the link to your linkedin profile is broken, kindly fix it.
- [x] Add a live demo link
- [ ] Add some validations, I mean if I click on submit button without writing anything on the form, it would not accept to save the book
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

Commit:
Add TSE changes

sergiocortessat commented:
Pull request 2

Hello TSE. I am sure there was a confusion with the last review. As many of the requirements were already implemented in the project such as displaying dynamically the books, a read button, and a delete button, CSS, scripts, and layouts separated. the readme does not have any of the highlights mentioned and the LinkedIn is working perfectly. therefore I request once again a review. I have modified some of the requirements which I was missing such as the constructor implemented using the arrow function and added a live demo, lock-files, and style outside of the index.html

Commit:
Add readme correction and live demo

Commit:
Add linters fix

Commit:
Add package-lock

Commit:
Add css for style of navbar

Commit:
Add css style

Commit:
Add fixing css

Commit:
Add styles css fix

MuriloRoque reviewed (approved):
## Status: Approved :green_circle: 

### Hello @sergiocortessat :wave: 

### Thank you for making all of the requested changes. Your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[![Twitter Follow](https://img.shields.io/twitter/follow/MuriloRoquePai1?label=Follow%20Murilo%20on%20Twitter&style=social)](https://twitter.com/MuriloRoquePai1)
